### PR TITLE
Xapid 994

### PR DESCRIPTION
**Pull request by haomingzhang**
add IT for generic replication
It will build transicator docker (if necessary), run dockers, run test and cleanup. 
Please use `make dockertest`.

### CODE REVIEW

File: managerInterfaces.go [hunk 1]
```diff
@@ -8,7 +8,7 @@ import (
 type tokenManager interface {
 	getBearerToken() string
 	invalidateToken() error
-	getToken() *oauthToken
+	getToken() *OauthToken
```
Comment (sramamoorthy73) [inline managerInterfaces.go]: Is this needed because you plan to use it outside the package?

File: dockertests/docker_test.go [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+
+			createTestTable(tableName);
+
+
+		}, 5)
+	})
+})
+
+func createTestTable(tableName string) {
+	tx, err := pgManager.BeginTransaction()
```
Comment (sramamoorthy73) [inline dockertests/docker_test.go]: If possible add enum data types as well and ensure it gets replicated in apid.

File: managerInterfaces.go (hunk 1 above)
Comment (haomingzhang) [inline managerInterfaces.go]: Yes. In fact my docker test is in a separate package. I write a simple dummy token auth server to make apid happy.

File: dockertests/docker_test.go (hunk 2 above)
Comment (haomingzhang) [inline dockertests/docker_test.go]: That's a good point! I'll add it now.

**Pull request merged**